module/zfs/arc.c

```diff
@@ -2071,6 +2097,15 @@ arc_buf_fill(arc_buf_t *buf, spa_t *spa, const zbookmark_phys_t *zb,
 		if (ARC_BUF_SHARED(buf)) {
 			ASSERT(arc_buf_is_shared(buf));
 		} else {
+			if (hdr->b_l1hdr.b_pabd == NULL) {
+				zfs_dbgmsg("hdr pabd is NULL! compress=%d "
+				    "encrypted=%d psize=%d lsize=%d "
+				    "objset=%llu object=%llu",
+				    arc_hdr_get_compress(hdr), encrypted,
+				    HDR_GET_PSIZE(hdr), HDR_GET_LSIZE(hdr),
+				    (u_longlong_t)zb->zb_objset,
+				    (u_longlong_t)zb->zb_object);
+			}
```

Comment:
Return error here? Otherwise its going to immediately panic when `abd_copy_to_buf` derefs `b_pabd`.

Reply:
Oh never mind, maybe your purpose is to just report more when it does crash?